### PR TITLE
Support mosn as a sidecar

### PR DESCRIPTION
**Pull request by wangfakang**
Now MOSN could run pass bookinfo case in istio-1.4.6. Use as follows，

### Render the  Istio yaml 

```bash
git clone  -b feature-mosn_adapter  https://github.com/mosn/istio.git &  cd istio 

helm template install/kubernetes/helm/istio-init --set global.hub=docker.io/istio --set global.tag=1.4.6  --name istio-init --namespace istio-system >istio_init.yaml

helm template install/kubernetes/helm/istio --set global.sidecar.binaryPath=/usr/local/bin/mosn  --set global.hub=docker.io/istio --set global.tag=1.4.6  --set global.sidecar.image=mosnio/proxyv2:1.4.6-mosn  --name istio --namespace istio-system >istio.yaml
```

### Install Istio with kubectl

```bash
kubectl create namespace istio-system
kubectl apply -f istio_init.yaml
kubectl apply -f istio.yaml
```

A few minutes later, you can execute `kubectl get pod -n istio-system` and see something like follows. Then you can run the bookinfo example by [bookinfo doc](https://mosn.io/zh/docs/quick-start/istio/#bookinfo-%E5%AE%9E%E9%AA%8C).

```
NAME                                        READY   STATUS      RESTARTS   AGE
istio-citadel-6b7796cbc7-8q72s              1/1     Running     0          11h
istio-galley-5545dff574-6qls6               1/1     Running     0          11h
istio-ingressgateway-79b75c4db-8w646        1/1     Running     0          59m
istio-init-crd-10-1.4-dev-9m7g6             0/1     Completed   0          11h
istio-init-crd-11-1.4-dev-j27vt             0/1     Completed   0          11h
istio-init-crd-14-1.4-dev-4h9ts             0/1     Completed   0          11h
istio-pilot-6654897fc6-vh299                2/2     Running     0          10h
istio-policy-68774796b8-qq5kz               2/2     Running     0          9h
istio-security-post-install-1.4-dev-g98ws   0/1     Completed   0          11h
istio-sidecar-injector-f7b64d984-hw8p4      1/1     Running     0          11h
istio-telemetry-5f5d94c78-zw847             2/2     Running     0          9h
prometheus-6c9c8f9c97-mkcnp                 1/1     Running     0          11h
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -22,6 +22,12 @@
 # allow optional per-repo overrides
 -include Makefile.overrides.mk
 
+
+# Dynamic set sidcar
+# Now it support Envoy and MOSN, if set SIDCAR = MOSN, MOSN will be used as the proxy in Istio.
+export SIDCAR = MOSN
```
Comment (rootsongjc) [inline Makefile]: SIDECAR -> sidecar
All the words were misspelled in the file.

Comment (wangfakang) [reply Makefile]: done.

**Comment by rootsongjc** [comment]
Follow the guide to setup istio with MOSN inside, it works.

image

image

**Review by rootsongjc — APPROVED** [review]
/LGTM

**Pull request merged**